apps/backoffice-v2/src/pages/Entity/components/Case/hooks/usePendingRevisionEvents/utils/calculate-pending-workflow-events.ts

```diff
@@ -0,0 +1,27 @@
+import { TWorkflowById } from '@/domains/workflows/fetchers';
+import { StateTag } from '@ballerine/common';
+import { calculateWorkflowRevisionableEvent } from '@/pages/Entity/components/Case/hooks/usePendingRevisionEvents/utils/calculate-workflow-revisionable-event';
+import { IPendingEvent } from '@/pages/Entity/components/Case/hooks/usePendingRevisionEvents/interfaces';
+
+export const calculatePendingWorkflowEvents = (workflow: TWorkflowById): Array<IPendingEvent> => {
+  return workflow.context.documents
```

Comment:
Consider `Array.prototype.reduce` for similar logic. It's possible to create an array of items matching the specified conditions.

Reply:
true you should have filter first before the map.